src/EditorFeatures/Core/Extensibility/NavigationBar/NavigationBarPresentedItem.cs

```diff
@@ -2,35 +2,38 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Immutable;
-using Microsoft.VisualStudio.Text;
+using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.Editor
 {
     /// <summary>
     /// The items that are displayed in the Navigation Bar when it is not expanded. They are never
     /// indented and cannot be used as the target of navigation.
     /// </summary>
-    internal class NavigationBarPresentedItem : NavigationBarItem
+    // We suppress this as this type *does* override ComputeAdditionalHashCodeParts
+    internal class NavigationBarPresentedItem : NavigationBarItem, IEquatable<NavigationBarPresentedItem>
     {
         public NavigationBarPresentedItem(
             string text,
             Glyph glyph,
-            ImmutableArray<ITrackingSpan> trackingSpans,
-            ITrackingSpan? navigationTrackingSpan,
+            ImmutableArray<TextSpan> spans,
+            TextSpan? navigationSpan,
             ImmutableArray<NavigationBarItem> childItems,
             bool bolded,
             bool grayed)
-            : base(
-                  text,
-                  glyph,
-                  trackingSpans,
-                  navigationTrackingSpan,
-                  childItems,
-                  indent: 0,
-                  bolded: bolded,
-                  grayed: grayed)
+            : base(text, glyph, spans, navigationSpan, childItems, indent: 0, bolded: bolded, grayed: grayed)
         {
         }
+
+        public override bool Equals(object? obj)
+            => Equals(obj as NavigationBarPresentedItem);
+
+        public bool Equals(NavigationBarPresentedItem? other)
+            => base.Equals(other);
+
+        public override int GetHashCode()
+            => throw new NotImplementedException();
```

Comment:
GetHashCode isn't used in any of these codepaths (we're not putting these items in collections, we're just comparing the previous/current computed item to see if it changed).  As such i didn't bother implementing these.

Reply:
This seems like a trap for the future, but I don't feel too strongly about it. I guess anyone who does try to add these to a collection will find out about it pretty quickly.

Why make it abstract though, only to throw an exception in every implementation? seems like you already did half of the work necessary 😛